### PR TITLE
feat: Update templates to use Microsoft Logging 5.0.0

### LINKED ISSUE #3624: Update Uno.Core dependency Microsoft.Extensions.Logging to version 3+
## What would you like to be added:
`Uno.Core` depends on `Microsoft.Extensions.Logging` 1.1.1 but must depend on 3.1.6

## Why is this needed:
Modern `Microsoft.Extensions.Hosting` (version 3+) allows building a host similar to ASP.NET Core in any app (UWP, Xamarin, WPF etc.) with all the logging and dependency injection. See my sample at https://andrii.visualstudio.com/Elmish.Uno
But having a dependency on the old API (version 1.1.1) of `Microsoft.Extensions.Logging` requires to implement workarounds like
``` C#
var loggerFactory = serviceProvider.GetRequiredService<ILoggerFactory>();
var _loggerFactoryField =
    typeof(global::Uno.Extensions.LogExtensionPoint)
        .GetField("_loggerFactory", System.Reflection.BindingFlags.NonPublic
                                    | System.Reflection.BindingFlags.Static);
_loggerFactoryField.SetValue(null, loggerFactory);
```
And I assume can cause issues on AOT and WASM platforms.

## Affected packages
**Uno.Core 2.0.0**

## For which Platform:

- [X] iOS
- [X] Android
- [X] WebAssembly
- [X] WebAssembly renderers for Xamarin.Forms
- [X] macOS
- [X] Windows
- [X] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): fixes #3624, fixes https://github.com/unoplatform/uno/issues/4767, fixes https://github.com/unoplatform/uno/issues/3624, fixes https://github.com/unoplatform/uno/issues/2013

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the new behavior?

Templates now use updated Microsoft loggers to 5.0.0.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SamplesApp/SamplesApp.UITests/Microsoft_UI_Xaml_Controls/NavigationViewTests/NavigationView_Tests.cs [hunk 1]
```diff
@@ -15,6 +15,7 @@ public class NavigationView_Tests : SampleControlUITestBase
 	{
 		[Test]
 		[AutoRetry()]
+		[ActivePlatforms(Platform.iOS, Platform.Android)] // fails on WebAssembly
```
Comment (MartinZikmund) [inline src/SamplesApp/SamplesApp.UITests/Microsoft_UI_Xaml_Controls/NavigationViewTests/NavigationView_Tests.cs:18]: Should we create an issue for it?

File: src/SolutionTemplate/UnoSolutionTemplate/iOS/UnoQuickStart.iOS.csproj [hunk 2]
```diff
@@ -117,8 +117,8 @@
 	<ItemGroup>
 		<PackageReference Include="Uno.UI" Version="2.2.0" />
 		<PackageReference Include="Uno.UI.RemoteControl" Version="2.2.0" Condition="'$(Configuration)'=='Debug'" />
-		<PackageReference Include="Microsoft.Extensions.Logging.Console" Version="1.1.1" />
-		<PackageReference Include="Microsoft.Extensions.Logging.Filter" Version="1.1.1" />
+		<PackageReference Include="Microsoft.Extensions.Logging" Version="5.0.0" />
+    <PackageReference Include="Uno.Extensions.Logging.OSLog " Version="1.0.0-dev.3" />
```
Comment (MartinZikmund) [inline src/SolutionTemplate/UnoSolutionTemplate/iOS/UnoQuickStart.iOS.csproj:121]: Why is the difference on iOS 😮 ?

**Review by MartinZikmund — APPROVED** [review]
🎢

File: src/SamplesApp/SamplesApp.UITests/Microsoft_UI_Xaml_Controls/NavigationViewTests/NavigationView_Tests.cs (hunk 1 above)
Comment (jeromelaban) [reply src/SamplesApp/SamplesApp.UITests/Microsoft_UI_Xaml_Controls/NavigationViewTests/NavigationView_Tests.cs:18]: I'm looking at it right now

File: src/SolutionTemplate/UnoSolutionTemplate/iOS/UnoQuickStart.iOS.csproj (hunk 2 above)
Comment (jeromelaban) [reply src/SolutionTemplate/UnoSolutionTemplate/iOS/UnoQuickStart.iOS.csproj:121]: iOS logging though the console does not go to the system log anymore, which is generally a problem

**Review by carldebilly — APPROVED** [review]

**Comment by xperiandri** [comment]
🍾🎉

**Pull request merged**